docs/src/literate-tutorials/hyperelasticity.jl

```diff
@@ -400,12 +397,11 @@ function solve()
         end
 
         ## Compute increment using conjugate gradients
-        @timeit "linear solve" IterativeSolvers.cg!(ΔΔu, K, g; maxiter=1000)
-
-        apply_zero!(ΔΔu, dbcs)
-        Δu .-= ΔΔu
```

Comment:
Imho it would be nice if the example would include load stepping.

Reply:
I'm happy to revert the change, but why is the original better? to me, it just appears to be more complicated.

Reply:
I agree that load stepping would be nice, but it  is a different issue. I would suggest postpone the load stepping into a separate PR to make a how-to via e.g. https://bifurcationkit.github.io/BifurcationKitDocs.jl/stable/IntroContinuation/ to also have proper adaptivity and possibly bifurcation detection.

Reply:
AFAIU the pedagogical aspect becomes more prominent if load stepping is included (linear ramping or adaptive stepping, which is sufficient for a vast majority of problems and no need for continuation methods). I agree that applying proper continuation methods should be a different tutorial

Reply:
For a time step, at the end of your Newton iteration you get the increment to your current solution (Δu) and each linear step gives the increment to that update (`ΔΔu`).

In my opinion it isn't "clean" to mutate the solution itself inside the Newton loop because:

- It isn't a solution to your problem.
- You might want to compare things to the previous solution, for example in plasticity you often compute deltas from the previous converged values, I've seen many people compute this deltas from the previous newton iteration.
- It makes it more clear that you should use `apply_zero!` for the boundary conditions (there is zero `Δu` on the dofs that have boundary conditions).

Yes, this would be more clear if one has some timestepping in this problem (e.g. to ramp the load).

Reply:
Also, I think it is just a good idea in general to keep a PR like this isolated to the important change/addition it is introducing.